zingolib/src/lightclient.rs

```diff
@@ -952,65 +957,81 @@ impl LightClient {
         let unified_spend_capability = &unified_spend_capability_arc.read().await;
-        let mut transaction_list = self
+        let mut consumer_ui_notes = self
             .wallet
             .transaction_context.transaction_metadata_set
             .read()
             .await
             .current
             .iter()
             .flat_map(|(_k, wallet_transaction)| {
-                let mut transactions: Vec<JsonValue> = vec![];
+                let mut consumer_notes_by_tx: Vec<JsonValue> = vec![];
 
+                let total_transparent_received = wallet_transaction.received_utxos.iter().map(|u| u.value).sum::<u64>();
                 if wallet_transaction.is_outgoing_transaction() {
-                    // If money was spent, create a transaction. For this, we'll subtract
+                    // If money was spent, create a consumer_ui_note. For this, we'll subtract
                     // all the change notes + Utxos
-                    transactions.push(Self::append_change_notes(wallet_transaction, include_memo_hex));
+                    // TODO:  Figure out why we have an insane comment saying that we "create a transaction"
+                    // in the middle of the "list transactions" fn/
+                    consumer_notes_by_tx.push(Self::append_change_notes(wallet_transaction, include_memo_hex, total_transparent_received));
                 }
 
-                // For each note that is not a change, add a transaction.
-                transactions.extend(self.add_wallet_notes_in_transaction_to_list(wallet_transaction, &include_memo_hex, unified_spend_capability));
+                // For each note that is not a change, add a consumer_ui_note.
+                consumer_notes_by_tx.extend(self.add_nonchange_notes(wallet_transaction, &include_memo_hex, unified_spend_capability));
 
+                // TODO:  determine if all notes are either Change-or-NotChange, if that's the case
+                // add a sanity check that asserts all notes are processed by this point
+                
                 // Get the total transparent value received in this transaction
-                let total_transparent_received = wallet_transaction.utxos.iter().map(|u| u.value).sum::<u64>();
-                let wallet_transparent_value_delta = total_transparent_received as i64 - wallet_transaction.total_transparent_value_spent as i64;
-                let address = wallet_transaction.utxos.iter().map(|utxo| utxo.address.clone()).collect::<Vec<String>>().join(",");
-                if wallet_transparent_value_delta > 0 {
-                    if let Some(transaction) = transactions.iter_mut().find(|transaction| transaction["txid"] == wallet_transaction.txid.to_string()) {
+                // Again we see the assumption that utxos are incoming.
+                let transparent_received = total_transparent_received as i64 - wallet_transaction.total_transparent_value_spent as i64;
```

Comment:
Do we know that `total_transparent_received => wallet_transaction.total_transparent_value_spent`? `wallet_transparent_value_delta` might be the better name if this could be negative, which I think it could be.